src/util/delay.test.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-env jest */
+
+import delay from './delay'
+jest.useFakeTimers()
+
+describe('delay', () => {
+    test('should return with the given delay', () => {
+        delay(1000)
+        expect(setTimeout.mock.calls.length).toBe(1)
+        expect(setTimeout.mock.calls[0][1]).toBe(1000)
+    })
+
+    test('should return a promise which resolves', () => {
```

Comment:
Actually that was the structure followed by me initially. But for some reason the test just doesn't pass if the expect is at the end of the test. I currently have no explanation for it, but I read on a blog that this works for a guy.